splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java

```diff
@@ -424,6 +435,124 @@ public static void DROP_TABLE_STATISTICS(String schema, String table) throws SQL
         }
     }
 
+    public static void FAKE_TABLE_STATISTICS(String schema,
+                                             String table,
+                                             long rowCount,
+                                             int meanRowWidth,
+                                             long numPartitions,
+                                             ResultSet[] outputResults) throws SQLException {
+
+        EmbedConnection conn = (EmbedConnection) SpliceAdmin.getDefaultConn();
+        try {
+            schema = EngineUtils.validateSchema(schema);
+            table = EngineUtils.validateTable(table);
+            TableDescriptor tableDesc = verifyTableExists(conn, schema, table);
+
+            if (rowCount < 0)
+                throw ErrorState.LANG_INVALID_FAKE_STATS.newException("table", "row count cannot be a negative value");
+
+            if (meanRowWidth <= 0)
+                throw ErrorState.LANG_INVALID_FAKE_STATS.newException("table", "meanRowWidth has to be greater than 0");
+            if (numPartitions <= 0)
+                throw ErrorState.LANG_INVALID_FAKE_STATS.newException("table", "number of partitions has to be greater than 0");
+
+            List<TableDescriptor> tds = Collections.singletonList(tableDesc);
+            authorize(tds);
+            DataDictionary dd = conn.getLanguageConnection().getDataDictionary();
+            dd.startWriting(conn.getLanguageConnection());
+            TransactionController tc = conn.getLanguageConnection().getTransactionExecute();
+            tc.elevate("statistics");
+            dropTableStatistics(tds,dd,tc);
+            ddlNotification(tc, tds);
+
+            // compose the fake table stats row
+            ExecRow statsRow;
+            int statsType = SYSTABLESTATISTICSRowFactory.FAKE_MERGED_STATS;
+            long conglomerateId = tableDesc.getHeapConglomerateId();
+
+            statsRow = StatisticsAdmin.generateRowFromStats(conglomerateId, "-All-", rowCount, rowCount*meanRowWidth, meanRowWidth, numPartitions, statsType, 0.0d);
```

Comment:
Do we maybe want sample fraction to be 1 instead of zero?  Otherwise, totalCount estimates might get adjusted down to zero in other locations?
`        if (isSampleStats)
            totalCount = (long)((double)totalCount/sampleFraction);`

Reply:
This is consistent with the setting in StatisticsAdmin.COLLECT_TABLE_STATISTICS(). As sample stats were added later, to be compatible with existing stats entries, the default value was set to be 0. So I prefer not to change the behavior in this JIRA. `sampleFraction` is only meaningful when `isSampleStats` is true.